pr_toolbox/quantum/results/frequencies.py

```diff
@@ -38,27 +60,61 @@ def map_counts(counts: Counts, mapper: Callable) -> Counts:
     return Counts(counts_dict)
 
 
-def bitflip_counts(counts: Counts, bitflips: int) -> Counts:
-    """Flip readout bits in counts according to the input bitflips (int encoded).
+def map_quasi_dists(quasi_dists: QuasiDistribution, mapper: Callable) -> QuasiDistribution:
+    """Map quasi-distributions by reassigning keys according to input callable.
 
     Args:
-        counts: the counts to process.
+        quasi_dists: the quasi-distributions to process.
+        mapper: the callable to map readout bits (i.e. counts keys).
+
+    Returns:
+        New QuasiDistribution with readout bits mapped according to input callable.
+    """
+    counts_dict: dict[int, float] = defaultdict(lambda: 0)
+    for readout, freq in quasi_dists.items():
+        readout = mapper(readout)
+        counts_dict[readout] += freq
```

Comment:
I wonder if it would make sense having a function that does this for a `dict` type directly. This would spare us some code duplicity.

Single dispatch could then call itself recursively, meaning that if it receives `Counts` it will convert them to a frequencies dict (i.e. `counts.int_outcomes()` and call itself again with this new argument type. The result will then be wrapped in a `Counts` object and returned. Something similar would happen for `QuasiDistribution`.